gpii/node_modules/orca/orcaSettingsHandler.js

```diff
@@ -78,104 +53,206 @@
+                };
+                togo.reject(rej);
+            }
+        };
+
+        try {
+            fs.statSync(orcaSettingsFile);
+            performRead();
+        } catch (err) {
+            var makeConfigurable = that.makeConfigurable();
+            makeConfigurable.then(function () {
+                performRead();
+            }, function (err) {
+                togo.reject(err);
+            });
+        }
 
-        var newSettingsResponse = {};
-        fluid.each(settings, function (settingVal, settingKey) {
-            var oldValue = fluid.get(currentSettingsInProfile, settingKey,
-                fluid.model.escapedGetConfig);
-
-            var change = {
-                path: settingKey,
-                value: settingVal
-            };
-            change.type = settingVal === undefined ? "DELETE" : "ADD";
-            applier.fireChangeRequest(change);
-
-            newSettingsResponse[settingKey] = {
-                "oldValue": oldValue,
-                "newValue": settingVal
-            };
-
-            fluid.set(currentSettingsInProfile, settingKey, settings[settingKey],
-                      fluid.model.escapedSetConfig);
-        });
+        return togo;
+    };
 
-        userSettings.profiles[userSettings.general.activeProfile[PROFILE_ID]] = currentSettingsInProfile;
+    gpii.orca.settingsHandler.makeConfigurable = function (that) {
+        var togo = fluid.promise();
 
-        fs.writeFileSync(orcaSettingsFile,
-                         JSON.stringify(userSettings, null, 4));
+        var orcaSpawn = spawn("orca", ORCA_ARGS);
+        var pass = 0;
+        var maxPass = 10;
+        var err;
 
-        return newSettingsResponse;
-    }
+        var resolvePromise = function () {
+            orcaSpawn.kill("SIGKILL");
+            togo.resolve();
+        };
+
+        var checkIfFileGetsCreated = function () {
+            pass++;
+            if (pass === maxPass) {
+               orcaSpawn.kill("SIGKILL");
+               var err = "Time limit exceeded [" + maxPass * 500 +
+                         "ms] for creating Orca's configuration file";
+               var rej = {
+                   isError: true,
+                   message: err
+               };
+               togo.reject(rej);
+            } else {
+                try {
+                    fs.statSync(orcaSettingsFile);
+                    resolvePromise();
+                } catch (err) {
+                    gpii.invokeLater(checkIfFileGetsCreated, 500);
+                };
+            }
+        };
 
-    gpii.orca.getImpl = function (payload) {
-        var app = fluid.copy(payload);
+        gpii.invokeLater(checkIfFileGetsCreated, 500);
+        return togo;
+    };
 
-        var settings = app.settings;
+    gpii.orca.settingsHandler.getSettings = function (that, payload) {
+        var togo = fluid.promise();
+
+        var loadSettings = that.loadSettings();
```

Comment:
I don't think it helps much to make the settingsHandler a modelComponent.
Instead, rewrite loadSettings so that it yields the actual settings in its resolve(), then this block can just be
return fluid.promise.map(loadSettings, function (settings) {
    var profile = settings.general.activeProfile[PROFILE_ID];
etc.